dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsIT.java

```diff
@@ -0,0 +1,141 @@
+
+class MetricsIT extends BaseIT {
+
+    private MetricsDAO metricsDAO;
+    private Session session;
+    private WorkflowDAO workflowDAO;
+    private WorkflowVersionDAO workflowVersionDAO;
+
+    @BeforeEach
+    public void setup() {
+        DockstoreWebserviceApplication application = SUPPORT.getApplication();
+        SessionFactory sessionFactory = application.getHibernate().getSessionFactory();
+        this.metricsDAO = new MetricsDAO(sessionFactory);
+        this.workflowDAO = new WorkflowDAO(sessionFactory);
+        this.workflowVersionDAO = new WorkflowVersionDAO(sessionFactory);
+
+        // used to allow us to use DAOs outside of the web service
+        this.session = application.getHibernate().getSessionFactory().openSession();
+        ManagedSessionContext.bind(session);
+    }
+
+    @BeforeEach
+    @Override
+    public void resetDBBetweenTests() throws Exception {
+        CommonTestUtilities.cleanStatePrivate2(SUPPORT, false, testingPostgres);
+    }
+
+    /**
+     * Demos the creation, update, and persistence of the Metrics entity
+     */
+    @Test
+    void testDAOs() {
+        final Transaction transaction = session.beginTransaction();
+
+        Metrics metrics1 = createMetrics();
+        Metrics metrics2 = createMetrics();
+        metricsDAO.create(metrics1);
+        metricsDAO.create(metrics2);
+
+        // Create a workflow and workflow version so we can add metrics for a specific platform to the workflow version
+        Workflow workflow = createWorkflow();
+        workflowDAO.create(workflow);
+        WorkflowVersion workflowVersion = createWorkflowVersion(workflow);
+        workflowVersion.getMetricsByPlatform().put(Partner.TERRA, metrics1);
+        workflowVersion.getMetricsByPlatform().put(Partner.DNA_STACK, metrics2);
+        workflowVersionDAO.create(workflowVersion);
+
+        session.flush();
+        transaction.commit();
+        session.close();
+    }
+
+    private Workflow createWorkflow() {
+        Workflow workflow = new BioWorkflow();
+        workflow.setSourceControl(SourceControl.GITHUB);
+        workflow.setOrganization("dockstore");
+        workflow.setRepository("dockstore-ui2");
+        workflow.setWorkflowName("test");
+        workflow.setDescriptorType(DescriptorLanguage.CWL);
+        return workflow;
+    }
+
+    private WorkflowVersion createWorkflowVersion(Workflow parent) {
+        WorkflowVersion workflowVersion = new WorkflowVersion();
+        workflowVersion.setName("2.0.1");
+        workflowVersion.setReference("2.0.1");
+        workflowVersion.setWorkflowPath("github.com/foo/bar");
+        workflowVersion.setParent(parent);
+        return workflowVersion;
+    }
+
+    private Metrics createMetrics() {
+        Metrics metrics = new Metrics();
+
+        ExecutionStatusCountMetric executionStatusCountMetric = new ExecutionStatusCountMetric();
+        executionStatusCountMetric.addCount(ExecutionStatusCountMetric.ExecutionStatus.SUCCESSFUL, 10);
```

Comment:
let's add some javadoc comments here, the code is pretty terse (which is good), but it would be nice to have one tutorial example here

"this section adds 10 successful workflow runs and one failed run ... " etc.